src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/ContextStack.cs

```diff
@@ -0,0 +1,161 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+
+
+
+namespace System.ComponentModel.Design.Serialization
+{
+    using System;
+    using System.Collections;
+    using System.Security.Permissions;
+
+    /// <summary>
+    ///     A context stack is an object that can be used by serializers
+    ///     to push various context objects.  Serialization is often
+    ///     a deeply nested operation, involving many different 
+    ///     serialization classes.  These classes often need additional
+    ///     context information when performing serialization.  As
+    ///     an example, an object with a property named "Enabled" may have
+    ///     a data type of System.Boolean.  If a serializer is writing
+    ///     this value to a data stream it may want to know what property
+    ///     it is writing.  It won't have this information, however, because
+    ///     it is only instructed to write the boolean value.  In this 
+    ///     case the parent serializer may push a PropertyDescriptor
+    ///     pointing to the "Enabled" property on the context stack.
+    ///     What objects get pushed on this stack are up to the
+    ///     individual serializer objects.
+    /// </summary>
+    [HostProtection(SharedState = true)]
+    [System.Security.Permissions.PermissionSetAttribute(System.Security.Permissions.SecurityAction.LinkDemand, Name = "FullTrust")]
+    public sealed class ContextStack
+    {
+        private ArrayList _contextStack;
```

Comment:
Same general question as earlier about replacing the non-generic collections with generic ones wherever possible.